autosar/behavior.py

```diff
@@ -968,54 +968,61 @@ def createDataReceivedEvent(self, runnableName, dataElementRef, modeDependency=N
 
         runnable=self.find(runnableName)
         if runnable is None:
-            raise ValueError('invalid runnable name: '+runnableName)
+            raise autosar.base.InvalidRunnableRef(runnableName)
         assert(isinstance(runnable, autosar.behavior.RunnableEntity))
 
         if not isinstance(dataElementRef, str):
-            raise ValueError("expected dataElementRef to be string of the format 'portName/dataElementName' ")
+            raise autosar.base.InvalidDataElementRef("expected dataElementRef to be string of the format 'portName' or 'portName/dataElementName' ")
+
         parts = autosar.base.splitRef(dataElementRef)
         if len(parts)==2:
-            raise ValueError("expected dataElementRef to be string of the format 'portName/dataElementName' ")

```

Comment:
This line was the bug?

Reply:
Yes and also the check `if (dataElement is None) or not isinstance(dataElement, autosar.portinterface.Operation):` in two places.